tests/test_utils.py

```diff
@@ -193,6 +206,31 @@ def test_extract_model(self):
 
         assert model == model_unwrapped
 
+    @require_tpu
+    @require_huggingface_suite
+    def test_extract_model_recursive_fsdpv2(self):
```

Comment:
While we don't have runners for this test, @shub-kris verified it works
```
One more thing that needs to be in the test which I missed is: One has to make sure that you set this two environment variables:
export PJRT_DEVICE=TPU XLA_USE_SPMD=1
I encountered the error when I didn't set XLA_USE_SPMD=1
 Please enable SPMD via `torch_xla.runtime.use_spmd()`
```

Reply:
I will do a test-run on this https://github.com/huggingface/transformers/issues/29659 to make sure training and loading of model works properly.

Reply:
@muellerzr shouldn't we add that in the test function that it should be ran with 
```bash
export PJRT_DEVICE=TPU XLA_USE_SPMD=1
```

Reply:
> I will do a test-run on this https://github.com/huggingface/transformers/issues/29659 to make sure training and loading of model works properly.

Ahh I can't do as the changes aren't there yet in the `transformers` library.